debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/RecordsStreamProducer.java

```diff
@@ -230,8 +230,12 @@ private void closeConnections() {
     }
 
     private void process(ReplicationMessage message, Long lsn, BlockingConsumer<ChangeEvent> consumer) throws SQLException, InterruptedException {
+        // in some cases we can get null if PG gives us back a message earlier than the latest reported flushed LSN.
+        // WAL2JSON can also send empty changes for DDL, materialized views, etc. and the heartbeat still needs to fire.
         if (message == null) {
-            // in some cases we can get null if PG gives us back a message earlier than the latest reported flushed LSN
+            lastCompletelyProcessedLsn = lsn;
```

Comment:
I think we still want to set this but I am not entirely sure.

Reply:
Let's add a `TRACE` message so we know when this happens